htsohm/plotting/data_points.py

```diff
@@ -0,0 +1,153 @@
+        if 'surface_area' in properties:
+            sa_c_tuples = session.query(Material.sa_volumetric_surface_area) \
+                    .filter(*filters, Material.generation == generation).all()
+            c['sa'] = [e[0] for e in sa_c_tuples]
+
+            sa_p_tuples = session.query(Material.sa_volumetric_surface_area) \
+                    .filter(*filters, Material.generation < generation).all()
+            p['sa'] = [e[0] for e in sa_p_tuples]
+
+        if 'helium_void_fraction' in properties:
+            vf_c_tuples = session.query(Material.vf_helium_void_fraction) \
+                    .filter(*filters, Material.generation == generation).all()
+            c['vf'] = [e[0] for e in vf_c_tuples]
+
+            vf_p_tuples = session.query(Material.vf_helium_void_fraction) \
+                    .filter(*filters, Material.generation < generation).all()
+            p['vf'] = [e[0] for e in vf_p_tuples]
+
+    return data
+
+def get_limits(config, x):
+    if x == 'ga':
+        return config['gas_adsorption_0']['limits']
+    elif x == 'sa':
+        return config['surface_area']['limits']
+    elif x == 'vf':
+        return config['helium_void_fraction']['limits']
+    else:
+        print('Unexpected flag.')
+
+def get_label(x):
+    if x == 'ga':
+        return 'Volumetric gas adsorption ({0}/{0})'.format(r'$cm^3$')
+    elif x == 'sa':
+        return 'Volumetric surface area ({}/{})'.format(r'$m^2$', r'$cm^3$')
+    elif x == 'vf':
+        return 'Helium void fraction ({})'.format(r'$dim.$')
+    else:
+        print('Unexpected flag.')
+
+def plot_2D(run_id, generations):
+    htsohm_dir = os.path.dirname(os.path.dirname(htsohm.__file__))
+    config_path = os.path.join(htsohm_dir, run_id, 'config.yaml')
+    config = load_config_file(config_path)
+    properties = config['material_properties']
+
+    data_types = []
+    if 'helium_void_fraction' in properties:
+        data_types.append('vf')
+    if 'surface_area' in properties:
+        data_types.append('sa')
+    if 'gas_adsorption_0' in properties:
+        data_types.append('ga')
+    data_combinations = []
+    for i in range(len(data_types)):
+        for j in range(i + 1, len(data_types)):
+            combo = [data_types[i], data_types[j]]
+            data_combinations.append(combo)
+
+    data = query_data_points(run_id, generations)
+
+    fig = plt.figure(figsize=(4 * len(generations), 4 * len(data_combinations)))
+    G = gridspec.GridSpec(4 * len(data_combinations), 4 * len(generations))
+
+    col_counter = 0
+    for generation in generations:
+        d = data['gen_{}'.format(generation)]
+        c, p = d['current'], d['previous']
+
+        row_counter = 0
+        for combo in data_combinations:
+            ax = plt.subplot2grid(
+                    (4 * len(data_combinations), 4 * len(generations)),
+                    (row_counter, col_counter), rowspan = 4, colspan = 4)
+            plt.scatter(p[combo[0]], p[combo[1]], edgecolor = 'none',
```

Comment:
I just learned last week that Matplotlib has two interfaces: 1) a state machine like interface where everything is plt.this and plt.that and mimics the matlab interface, and 2) a more pythonic object-oriented interface where you get the figure via plt.figure() but then use the figure object to do everything else. See https://matplotlib.org/faq/usage_faq.html

I hate matplotlib MUCH less after finding the OO interface.